packages/typegpu/src/tgsl/generationHelpers.ts

```diff
@@ -245,7 +246,9 @@ export function getTypeFromWgsl(resource: Wgsl): AnyData | UnknownData {
     }
   }
 
-  return isWgslData(resource) ? resource : UnknownData;
+  return isWgslData(resource) || isSelfResolvable(resource)
```

Comment:
Could you jot down a comment saying that this is temporary, until we figure out textures-as-data?